GVFS/GVFS.FunctionalTests/Tests/GitCommands/GitCommandsTests.cs

```diff
@@ -415,110 +416,52 @@ public void CaseOnlyRenameFileAndChangeBranches()
-            this.ValidateGitCommand("checkout -b " + newBranchName);
-
-            // Confirm that no other test has caused "Protocol.md" to be added to the modified paths
-            string fileName = "Protocol.md";
-            GVFSHelpers.ModifiedPathsShouldNotContain(this.Enlistment, this.FileSystem, fileName);
-
-            string controlTargetFolder = "MoveFileFromInsideRepoToOutsideRepoAndCommit_ControlTarget";
-            string gvfsTargetFolder = "MoveFileFromInsideRepoToOutsideRepoAndCommit_GVFSTarget";
-
-            // Create the target folders in this.Enlistment.EnlistmentRoot as it's outside of src and the control repo
-            // and is cleaned up when the functional tests run
-            string controlTargetFolderPath = Path.Combine(this.Enlistment.EnlistmentRoot, controlTargetFolder);
-            string gvfsTargetFolderPath = Path.Combine(this.Enlistment.EnlistmentRoot, gvfsTargetFolder);
-            string controlTargetFilePath = Path.Combine(controlTargetFolderPath, fileName);
-            string gvfsTargetFilePath = Path.Combine(gvfsTargetFolderPath, fileName);
+            this.MoveFolderFromOutsideRepoToInsideRepoAndAdd(true);
+        }
 
-            // Move control repo file
-            this.FileSystem.CreateDirectory(controlTargetFolderPath);
-            this.FileSystem.MoveFile(Path.Combine(this.ControlGitRepo.RootPath, fileName), controlTargetFilePath);
-            controlTargetFilePath.ShouldBeAFile(this.FileSystem);
+        // Linux only because Linux uses a separate repo mount device
+        [TestCase]
+        [Category(Categories.RepositoryMountsDifferentFileSystem)]
+        public void MoveFolderFromOutsideRepoToInsideRepoOnDifferentFileSystemAndAdd()
+        {
+            this.MoveFolderFromOutsideRepoToInsideRepoAndAdd(false);
+        }
 
-            // Move GVFS repo file
-            this.FileSystem.CreateDirectory(gvfsTargetFolderPath);
-            this.FileSystem.MoveFile(Path.Combine(this.Enlistment.RepoRoot, fileName), gvfsTargetFilePath);
-            gvfsTargetFilePath.ShouldBeAFile(this.FileSystem);
+        // Mac and Windows only because Linux uses a separate repo mount device
+        [TestCase]
+        [Category(Categories.RepositoryMountsSameFileSystem)]
+        public void MoveFileFromInsideRepoToOutsideRepoOnSameFileSystemAndCommit()
+        {
+            this.MoveFileFromInsideRepoToOutsideRepoAndCommit(true);
+        }
 
-            this.ValidateGitCommand("status");
-            this.ValidateGitCommand("add .");
-            this.RunGitCommand("commit -m \"Change for MoveFileFromInsideRepoToOutsideRepoAndCommit\"");
-            this.ValidateGitCommand("checkout " + this.ControlGitRepo.Commitish);
+        // Linux only because Linux uses a separate repo mount device
+        [TestCase]
+        [Category(Categories.RepositoryMountsDifferentFileSystem)]
+        public void MoveFileFromInsideRepoToOutsideRepoOnDifferentFileSystemAndCommit()

```

Comment:
Since these kinds of activities are effectively impermissible on Linux, I think the same logic above holds and so I've just marked these are `RepositoryMountsSameFileSystem` and left it at that.